otx/cli/builder/builder.py

```diff
@@ -0,0 +1,276 @@
+                use_out_indices = True
+            if arg_key == "depth" and "arch" in required_args:
+                continue
+            if arg_value.default is inspect.Parameter.empty and arg_key not in required_args:
+                required_args.append(arg_key)
+                continue
+            # Update Backbone out_indices to defaults
+            if arg_key not in backbone_config and arg_key in ("out_indices"):
+                backbone_config[arg_key] = arg_value.default
+        parent_function = parent_function[0].__bases__
+
+    for arg in required_args:
+        if arg not in backbone_config and arg not in ("args", "kwargs", "self"):
+            missing_args.append(arg)
+    if len(missing_args) > 0:
+        if not skip_missing:
+            raise ValueError(
+                f"[otx build] {backbone_config['type']} requires the argument : {missing_args}"
+                f"\n[otx build] Please refer to {inspect.getfile(backbone_function)}"
+            )
+        for arg in missing_args:
+            if arg in required_option:
+                backbone_config[arg] = f"!!!SELECT_OPTION: {required_option[arg]}"
+            else:
+                backbone_config[arg] = "!!!!!!!!!!!INPUT_HERE!!!!!!!!!!!"
+        print(
+            f"[otx build] {backbone_config['type']} requires the argument : {missing_args}"
+            f"\n[otx build] Please refer to {inspect.getfile(backbone_function)}"
+        )
+    backbone_config["use_out_indices"] = use_out_indices
+    return missing_args
+
+
+def update_in_channel(model_config, out_channels):
+    """Update in_channel of head or neck."""
+    if hasattr(model_config.model, "neck"):
+        print(f"\tUpdate model.neck.in_channels: {out_channels}")
+        model_config.model.neck.in_channels = out_channels
+    elif hasattr(model_config.model, "bbox_head"):
+        raise NotImplementedError("This architecture currently does not support public backbone.")
+    elif hasattr(model_config.model, "decode_head"):
+        print(f"\tUpdate model.decode_head.in_channels: {out_channels}")
+        model_config.model.decode_head.in_channels = out_channels
+    elif hasattr(model_config.model, "head"):
+        print(f"\tUpdate model.head.in_channels: {out_channels}")
+        model_config.model.head.in_channels = out_channels
+
+
+class Builder:
+    """Class that implements a model templates registry."""
+
+    def build_task_config(self, task_type, model_type=None, workspace_path=None):
+        """Create OTX workspace with Template configs from task type.
+
+        task_type: The type of task want to get (str)
+        model_type: Specifies the template of a model (str)
+        workspace_path: This is the folder path of the workspace want to create (os.path)
+        """
+
+        # Create OTX-workspace
+        if workspace_path is None:
+            workspace_path = f"./otx-workspace-{task_type}"
+        os.makedirs(workspace_path, exist_ok=True)
```

Comment:
Is it safe re-using existed workspace? Many templates and configs will be replaced

Reply:
Changed the value of exist_ok to False so that it is not replaced in a situation where it already exists.